lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -81,10 +103,65 @@ def test_sort_order(self):
         response = self.api_response()
         for course_num in range(num_courses):
             self.assertEqual(
-                response.data[course_num]['course']['id'],  # pylint: disable=no-member
+                response.data[course_num]['course']['id'],
                 unicode(courses[num_courses - course_num - 1].id)
             )
 
+    @patch.dict(
+        settings.FEATURES, {'ENABLE_PREREQUISITE_COURSES': True, 'MILESTONES_APP': True, 'DISABLE_START_DATES': False}
+    )
+    def test_courseware_access(self):
+        seed_milestone_relationship_types()
+        self.login()
+
+        course_with_prereq = CourseFactory.create(start=self.YESTERDAY, mobile_available=True)
+        prerequisite_course = CourseFactory.create()
+        set_prerequisite_courses(course_with_prereq.id, [unicode(prerequisite_course.id)])
+
+        # Create list of courses with various expected courseware_access responses and corresponding expected codes
+        courses = [course_with_prereq,
+                   CourseFactory.create(start=self.TOMORROW, mobile_available=True),
+                   CourseFactory.create(visible_to_staff_only=True, mobile_available=True),
+                   CourseFactory.create(start=self.YESTERDAY, mobile_available=True, visible_to_staff_only=False)]
+
+        expected_error_codes = [MilestoneError().error_code,  # 'unfulfilled_milestones'
+                                StartDateError(self.TOMORROW).error_code,  # 'course_not_started'
+                                VisibilityError().error_code,  # 'not_visible_to_user'
+                                None]
+
+        # Enroll in all the courses
+        for course in courses:
+            self.enroll(course.id)
+
+        # Verify courses have the correct response through error code. Last enrolled course is first course in response
+        response = self.api_response()
+        for course_num in range(len(courses)):
+            result = response.data[course_num]['course']['courseware_access']  # pylint: disable=no-member
+            self.assertEqual(result['error_code'], expected_error_codes[::-1][course_num])
+
+            if result['error_code'] is not None:
+                self.assertFalse(result['has_access'])
+
+    @ddt.data(
+        (TOMORROW, ADVERTISED_START, ADVERTISED_START, "string"),
+        (TOMORROW, None, defaultfilters.date(TOMORROW, "DATE_FORMAT"), "timestamp"),
+        (DEFAULT_START_DATE, ADVERTISED_START, ADVERTISED_START, "string"),
+        (DEFAULT_START_DATE, None, None, "empty")
+    )
+    @ddt.unpack
+    @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
+    def test_start_type_and_display(self, start, advertised_start, expected_display, expected_type):
+        """
+        Tests that the correct start_type and start_display are returned in the case the course has not started
```

Comment:
It's preferred for docstrings to be <=80 columns in width, so I'd break this into 2 lines.